packages/pharos/src/components/sheet/pharos-sheet.ts

```diff
@@ -204,10 +207,9 @@ export class PharosSheet extends ScopedRegistryMixin(PharosElement) {
     }
   }
 
-  private _handleOverlayClickOnTouch(event: TouchEvent): void {
-    const sheetContent = this.shadowRoot?.querySelector(`.sheet__content`) as HTMLDivElement;
-    const overlayContent = this.shadowRoot?.querySelector(`.sheet__overlay`) as HTMLDivElement;
-    if (overlayContent.clientHeight - event.touches?.[0].clientY > sheetContent.clientHeight) {
+  private _handleOverlayInteraction(event: MouseEvent | TouchEvent): void {
+    const interactionY = event instanceof MouseEvent ? event.clientY : event.touches?.[0].clientY;
+    if (this._sheetOverlay.clientHeight - interactionY > this._sheetContent.clientHeight) {
```

Comment:
This is the key functional change; the rest here is refactor to reduce duplication of the query selectors for the sheet overlay and sheet content